config/cache.go

```diff
@@ -38,3 +40,36 @@ func (cache *StringCache) Put(key string, value string) {
 	cacheKey := fmt.Sprintf("%x", md5Sum)
 	cache.Cache[cacheKey] = value
 }
+
+// IAMRoleOptionsCache - cache for IAMRole options
+type IAMRoleOptionsCache struct {
+	Cache map[string]options.IAMRoleOptions
+	Mutex *sync.Mutex
+}
+
+// NewIAMRoleOptionsCache - create new cache for IAM roles
+func NewIAMRoleOptionsCache() *IAMRoleOptionsCache {
+	return &IAMRoleOptionsCache{
+		Cache: map[string]options.IAMRoleOptions{},
+		Mutex: &sync.Mutex{},
+	}
+}
+
+// Get - get cached value, md5 hash is used as key to have fixed length keys and avoid duplicates
+func (cache *IAMRoleOptionsCache) Get(key string) (options.IAMRoleOptions, bool) {
+	cache.Mutex.Lock()
+	defer cache.Mutex.Unlock()
+	md5Sum := md5.Sum([]byte(key))
```

Comment:
Nit: Should consider using `sha-256`. `md5` is known to have collisions, although in this case it's very unlikely